File: app/src/main/java/com/remotearthsolutions/expensetracker/services/InternetCheckerServiceImpl.java
package com.remotearthsolutions.expensetracker.services;

import android.app.Service;
import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class InternetCheckerServiceImpl implements InternetCheckerService {

    private Context context;

    public InternetCheckerServiceImpl(Context context) {
        this.context = context;
    }

    @Override
    public boolean isConnected() {
        ConnectivityManager connectivity = (ConnectivityManager)
                context.getSystemService(Service.CONNECTIVITY_SERVICE);

        if (connectivity != null) {
            NetworkInfo info = connectivity.getActiveNetworkInfo();
            if (info != null) {
                return info.getState() == NetworkInfo.State.CONNECTED;
            }
        }
        return false;
    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/services/InternetCheckerServiceImpl.kt
package com.remotearthsolutions.expensetracker.services

import android.app.Service
import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

class InternetCheckerServiceImpl(private val context: Context) :
    InternetCheckerService {
    override val isConnected: Boolean
        get() {
            val connectivity =
                context.getSystemService(Service.CONNECTIVITY_SERVICE) as ConnectivityManager
            val info = connectivity.activeNetworkInfo
            if (info != null) {
                return info.state == NetworkInfo.State.CONNECTED
            }
            return false
        }

}